cpp/open3d/geometry/TriangleMesh.cpp

```diff
@@ -77,30 +74,29 @@ TriangleMesh &TriangleMesh::operator+=(const TriangleMesh &mesh) {
     if (HasAdjacencyList()) {
         ComputeAdjacencyList();
     }
-    if (add_textures) {
+    if (mesh.HasTriangleUvs()) {
         size_t old_tri_uv_num = triangle_uvs_.size();
         triangle_uvs_.resize(old_tri_uv_num + mesh.triangle_uvs_.size());
```

Comment:
If `this` does not have any UV coordinates then `triangle_uvs_.size()` will be different from the total number of triangles in the resulting mesh. This would break with the assumption that `triangle_uvs_.size() == 3*number of triangles`